EarthTube Humidity Ratio Not Tracked Properly in Zone Moisture Balance

RKStrand opened the pull request:
Pull request overview
---------------------
This pull request addresses the defect contained in GitHub Issue #4671.  In this defect, the earth tube humidity ratio was updated but when it was different than the outdoor humidity ratio due to condensation in the earth tube this was not accounted for in the zone water vapor balance.  This work corrects this error and also implements three new output variables.  Some diffs in the output for simulations that contain earth tubes may occur if condensation is occurring inside the earth tube.  Basically, if the ground temperature at the depth of the tube is lower than the dew-point temperature of the air entering the earth tube, then condensation will occur. The modified code for this submission corrects this problem.

### Work Checklist
Add to this list or remove from it as applicable.  This is a simple templated set of guidelines.
 - [ x ] Title of PR should be user-synopsis style (clearly understandable in a standalone changelog context)
 - [ x ] At least one of the following appropriate labels must be added to this PR to be consumed into the changelog:
   - Defect: This pull request repairs a github defect issue #4671.

### Review Checklist
This will not be exhaustively relevant to every PR.
 - [ ] Code style (parentheses padding, variable names)
 - [ ] Functional code review (it has to work!)
 - [ ] If defect, results of running current develop vs this branch should exhibit the fix
 - [ ] CI status: all green or justified
 - [ ] Performance: CI Linux results include performance check -- verify this
 - [ ] Unit Test(s)
 - C++ checks:
   - [ ] Argument types
   - [ ] If any virtual classes, ensure virtual destructor included, other things
 - IDD changes:
   - [ ] Verify naming conventions and styles, memos and notes and defaults
   - [ ] Open windows IDF Editor with modified IDD to check for errors
   - [ ] If transition, add rules to spreadsheet
   - [ ] If transition, add transition source
   - [ ] If transition, update idfs
 - [ ] If new idf included, locally check the err file and other outputs
 - [ ] Documentation changes in place
 - [ ] Changed docs build successfully
 - [ ] ExpandObjects changes?
 - [ ] If output changes, including tabular output structure, add to output rules file for interfaces

RKStrand committed:
Addition of new variables

Addition of new reporting variables as well as new humidity balance
term from the earth tube routine

RKStrand committed:
New Variables Added to Input File

Added the new variables to the test suite input file for earth tubes

RKStrand committed:
Modification to moisture balance

Code in predictor/corrector was modified to correct account for the
humidity ratio of the air entering the zone from the earth tube as it
might be different than the outside air.  Also added another variable
to the input files in the test suite so that the impact of this change
can be tracked.

RKStrand committed:
Unit Test and Doc Additions

This commit includes the unit test addition, modifications to existing
unit tests as a result of changes made in the earth tube code in
EnergyPlus, and the documentation additions for the new output
variables.

nrel-bot-3 commented:
@RKStrand @lgentile it has been 15 days since this pull request was last updated.

nrel-bot-3 commented:
@RKStrand @lgentile it has been 14 days since this pull request was last updated.

rraustad commented on src/EnergyPlus/EarthTube.cc:
```diff
@@ -566,44 +576,98 @@ namespace EarthTube {
+		// na
+		
+		// DERIVED TYPE DEFINITIONS
+		// na
```
@RKStrand we have been deleting unused comments in headers. I suggest removal. Also, please post a plot of results so reviewers have an easier time of understanding these changes.

RKStrand replied:
@rraustad I have deleted the unused comments in the headers, resolved the merge conflicts, and committed the changes.  I am also attaching a graph showing the outdoor humidity ratio, the old zone humidity ratio (V8.7), the new zone humidity ratio (after the fix), and the outlet earth tube humidity ratio after the fix.  These are all for the first zone in the test suite example file (other two zones in this file exhibit same response to the change).  Hopefully this addresses everything you mentioned and this is now ready for merging.  Thanks!
[EarthTubeHumRatComp.pdf](https://github.com/NREL/EnergyPlus/files/1222301/EarthTubeHumRatComp.pdf)

RKStrand committed:
Merge branch 'develop' into 147589329-Earthtube-HumRat-Not-Tracked

# Conflicts:
#	src/EnergyPlus/ZoneTempPredictorCorrector.cc
#	tst/EnergyPlus/unit/EarthTube.unit.cc

RKStrand committed:
Review Comments and Merge Diffs

Merged with current develop and addressed review comments about
superfluous comment lines from old F90 format

rraustad commented:
@Myoldmopar @mjwitte something is not right here. After develop was merged, all recent changes are showing up on the Files compare tab. Does this mean develop was not correctly merged?

Myoldmopar commented:
![image](https://user-images.githubusercontent.com/849824/29273665-c0c9d688-80ca-11e7-9604-7bb06880fe90.png)

This branch is still 507 commits behind develop.  Something went awry.

RKStrand commented:
@Myoldmopar @rraustad @mjwitte Ok...I'm not sure how this is 507 commits behind develop.  I updated develop on my machine (using the GitHub Desktop for Mac app).  Then I switched to this branch and updated from develop.  I manually resolved the merge issues and then committed and synced.  That was all earlier this morning.  Now, when I go into desktop, I see that there are five more uncommitted changes though I didn't do anything and these didn't seem show up earlier.  It's also saying that there are more unresolved merge issues.  So, I guess I will have to look at this more, but I'm not sure why this happened.

RKStrand commented:
@Myoldmopar @rraustad @mjwitte Having said that, I am also concerned now that this are seriously messed up and it won't matter what I do to this because it is now on the messed up track and that doesn't seem to ever get resolved properly.  Do I need to "start over" with a new branch now?

Myoldmopar commented:
Don't yet. I'll be back at my computer in a little bit and I'll try to
resolve it.

Myoldmopar committed:
Merge develop; resolve conflicts

Myoldmopar commented:
OK resolved.  Do a Git Pull and it should pick up my merge commit and be good to go.  I don't know exactly what went wrong, but as long as it's a rare occasion, then it's ok.  I wish I could be more help on the Github Desktop client, but I don't know what exactly it is trying to do.  

Anyway, I tried to resolve the conflicts.  Most of them were pretty straightforward, but please check out [this file](https://github.com/NREL/EnergyPlus/pull/6184/files#diff-26a720ee34299122c810df4c89b63101).  I couldn't tell exactly what variables needed to be moved around.  

Carry on...

RKStrand commented:
@Myoldmopar Awesome! You are just awesome.  I chucked whatever things were showing up in the desktop app and downloaded your new merge.  As far as I can tell by looking through every change in your merge, it looks good.  Going to run a few tests and then give @rraustad the go ahead to look at this.  Thanks to both of you for your help!

RKStrand commented:
@Myoldmopar Looks great--everything as expected, results match what I when I committed this a while ago.  We are a go from my end--just the review to complete.

@rraustad I think this is ready now, thanks to Edwin.  Hopefully you won't see any conflicts or other merge problems and hopefully I have taken care of your comments.  I did attach a graph to a previous comment so that you can more easily take a look at what is happening.

rraustad commented on src/EnergyPlus/EarthTube.cc:
```diff
@@ -134,42 +115,27 @@ namespace EarthTube {
+		TotEarthTube = 0;
+		EarthTubeSys.deallocate();
+		ZnRptET.deallocate();
+	}
```
Just so you know, all those deallocates at the end of the units tests should go here in clear_state. Then there would be no need to deallocate them in unit tests.

rraustad commented:
I see nothing wrong with these changes. There is no defect file to go on so I am looking at the plot provided. Other than the sharp change in zone humrat when the ET turns on or off, the zone humrat is clearly impacted by the earth tube.

I'd say this looks good.

![earthtube](https://user-images.githubusercontent.com/5778156/29341514-c2f1e3c8-81f2-11e7-979a-df1548bb5870.png)

rraustad commented:
Reviewed and merged.
